src/components/EnterpriseList/index.jsx

```diff
@@ -0,0 +1,166 @@
+import StatusAlert from '../StatusAlert';
+import LoadingMessage from '../LoadingMessage';
+import TableWithPagination from '../TableWithPagination';
+
+class EnterpriseList extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const { enterprises } = props;
+
+    this.state = {
+      columns: [
+        {
+          label: 'Enterprise',
+          key: 'link',
+          columnSortable: true,
+        },
+      ],
+      enterprises: this.formatEnterpriseData(enterprises),
+      pageCount: enterprises ? enterprises.num_pages : null,
+    };
+
+    this.renderTableContent = this.renderTableContent.bind(this);
+  }
+
+  componentDidMount() {
+    this.clearPortalConfiguration();
+    this.getEnterpriseList();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { enterprises } = this.props;
+
+    if (enterprises !== prevProps.enterprises) {
+      this.setState({ // eslint-disable-line react/no-did-update-set-state
+        enterprises: this.formatEnterpriseData(enterprises),
+        pageCount: enterprises ? enterprises.num_pages : null,
+      });
+    }
+  }
+
+  getEnterpriseList(options) {
+    this.props.getEnterpriseList(options);
+  }
+
+  clearPortalConfiguration() {
+    this.props.clearPortalConfiguration();
+  }
+
+  formatEnterpriseData(enterprises) {
+    if (!enterprises) {
+      return null;
+    } else if (!enterprises.results.length) {
+      return [];
+    }
+
+    return enterprises.results.map(enterprise => ({
+      link: (<Link to={`/${enterprise.slug}/admin`}>{enterprise.name}</Link>),
+      name: enterprise.name,
+      slug: enterprise.slug,
+      uuid: enterprise.uuid,
+    }));
+  }
+
+  renderEmptyEnterpriseListMessage() {
+    return (
+      <StatusAlert
+        alertType="warning"
+        iconClassName={['fa', 'fa-exclamation-circle']}
+        message="You do not have permission to view any enterprise customer data."
+      />
+    );
+  }
+
+  renderErrorMessage() {
+    return (
+      <StatusAlert
+        alertType="danger"
+        iconClassName={['fa', 'fa-times-circle']}
+        title="Unable to load enterprise list"
+        message={`Try refreshing your screen (${this.props.error.message})`}
+      />
+    );
+  }
+
+  renderRedirectToEnterpriseAdminPage() {
+    const { enterprises } = this.state;
+    return (
+      <Redirect to={`/${enterprises[0].slug}/admin`} />
+    );
+  }
+
+  renderTableContent() {
+    const {
+      columns,
+      enterprises,
+      pageCount,
+    } = this.state;
+
+    return (
+      <TableWithPagination
+        columns={columns}
+        data={enterprises}
+        pageCount={pageCount}
+        paginationLabel="enterprise list pagination"
+        tableSortable
+        handleDataUpdate={options =>
+          this.getEnterpriseList(options)
+        }
+      />
+    );
+  }
+
+  renderLoadingMessage() {
+    return <LoadingMessage className="enterprise-list" />;
+  }
+
+  render() {
+    const { className, loading, error } = this.props;
```

Comment:
I dont think className is needed as a props here. Doesn't look like it gets set and doesn't seem necessary